compiler/rustc_middle/src/ty/util.rs

```diff
@@ -828,6 +832,61 @@ impl<'tcx> ty::TyS<'tcx> {
         }
     }
 
+    pub fn gc_layout(&'tcx self, tcx: TyCtxt<'tcx>, param_env: ty::ParamEnv<'tcx>) -> (u64, u64) {
+        let layout = tcx.layout_of(param_env.and(self)).unwrap();
+        let w_size = Size::from_bytes(size_of::<usize>());
+        let w_align = Align::from_bytes(w_size.bytes()).unwrap();
+
+        if layout.align.abi.bytes() < (w_size.bytes())
+            || self.is_no_trace(tcx.at(DUMMY_SP), param_env)
+        {
+            // If an ADT is aligned to a size smaller than 1 word, then it
+            // contains no fields large enough to represent a GC pointer.
+            //
+            // By default, the collector will conservatively scan a struct's
+            // memory, treating it as word-sized chunks. However, since rust is
+            // free to shuffle field order around, any in-memory words which
+            // represent actual pointers would be purely accidental. Such
+            // structs should be ignored completely by the collector.
+            return (0, 0);
+        }
+
+        let src_field_iter = self.ty_adt_def().unwrap().all_fields();
+        let mut bitmap: u64 = u64::MAX;
```

Comment:
Is a bitmap 64-bits all the time, no matter the platform's word size?

Reply:
Yes

Reply:
OK.